rclpy/test/test_executor.py

```diff
@@ -302,6 +302,47 @@ def test_executor_add_node(self):
         assert not executor.add_node(self.node)
         assert id(executor) == id(self.node.executor)
 
+    def test_executor_spin_until_future_complete(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        def timer_callback():
+            pass
+        timer = self.node.create_timer(0.1, timer_callback)
+
+        from rclpy.task import Future
+
+        # Timeout
+        future = Future()
+        self.assertFalse(future.done())
+        start = time.monotonic()
+        executor.spin_until_future_complete(future=future, timeout_sec=1.0)
+        end = time.monotonic()
+        self.assertGreaterEqual(end - start, 1.0)
+        self.assertFalse(future.done())
+
+        # Future complete
+        future = Future()
+
+        def set_future_result():
+            nonlocal future
+            time.sleep(1.0)
```

Comment:
Can we remove this sleep? It would make for a faster test.

Reply:
That sleep is intended to test the situation that future completes in the middle of the spin. So, I'd not like to remove sleep. Instead, decrease it from 1.0s to 0.1s. Is it ok?

Reply:
Yeah, that's good.